v1/src/main/java/com/google/cloud/teleport/templates/SpannerVectorEmbeddingExport.java

```diff
@@ -0,0 +1,314 @@
+        helpText =
+            "Comma separated list of columns which are required for Vertex AI Vector Search Index."
+                + " If the column names don't precisely align with the Vertex AI Vector Search Index input structure, you can establish column mappings using aliases."
+                + " For e.g. if you have columns id and my_embedding i.e. the id column matches what vertex expects but the embedding column is named differently,"
+                + " you can specify the following `id, my_embedding:embedding`.")
+    ValueProvider<String> getSpannerColumnsToExport();
+
+    void setSpannerColumnsToExport(ValueProvider<String> value);
+
+    @TemplateParameter.GcsWriteFolder(
+        order = 60,
+        description = "Output files folder in Cloud Storage",
+        helpText = "The Cloud Storage folder for writing output files. Must end with a slash.",
+        example = "gs://your-bucket/folder1/")
+    ValueProvider<String> getGcsOutputFolder();
+
+    void setGcsOutputFolder(ValueProvider<String> value);
+
+    @TemplateParameter.Text(
+        order = 70,
+        description = "Output files prefix in Cloud Storage",
+        helpText = "The filename prefix for writing output files.",
+        example = "vector-embeddings")
+    ValueProvider<String> getGcsOutputFilePrefix();
+
+    void setGcsOutputFilePrefix(ValueProvider<String> textWritePrefix);
+
+    @TemplateParameter.Text(
+        order = 80,
+        optional = true,
+        description = "Cloud Spanner Endpoint to call",
+        helpText =
+            "The Cloud Spanner endpoint to call in the template. The default is set to https://batch-spanner.googleapis.com.",
+        example = "https://batch-spanner.googleapis.com")
+    @Default.String("https://batch-spanner.googleapis.com")
+    ValueProvider<String> getSpannerHost();
+
+    @SuppressWarnings("unused")
+    void setSpannerHost(ValueProvider<String> value);
+
+    @TemplateParameter.Text(
+        order = 90,
+        optional = true,
+        regexes = {
+          "^([0-9]{4})-([0-9]{2})-([0-9]{2})T([0-9]{2}):([0-9]{2}):(([0-9]{2})(\\.[0-9]+)?)Z$"
+        },
+        description = "Timestamp to read stale data from a version in the past.",
+        helpText =
+            "If set, specifies the time when the database version must be taken."
+                + " String is in the RFC 3339 format in UTC time. "
+                + " Timestamp must be in the past and maximum timestamp staleness applies; see "
+                + "<a href=\"https://cloud.google.com/spanner/docs/timestamp-bounds#maximum_timestamp_staleness\">Maximum Timestamp Staleness</a>."
+                + " If not set, strong bound is used to read the latest data; see "
+                + "<a href=\"https://cloud.google.com/spanner/docs/timestamp-bounds#strong\">Timestamp Strong Bounds</a>.",
+        example = "1990-12-31T23:59:60Z")
+    @Default.String(value = "")
+    ValueProvider<String> getSpannerVersionTime();
```

Comment:
Earlier we had it as `snapshotTime` only but as part of internal review we changed it to `versionTime` because `snapshotTime`  can be mistaken for a snapshot feature which we do not support.